qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -0,0 +1,203 @@
+                "grouped_property and reference_energy will not be set."
+            )
+            return
+
+        self._reference_energy = electronic_energy.reference_energy if not None else 0.0
+        self._grouped_property = grouped_property
+
+    @property
+    def ansatz(self) -> UCC:
+        """The UCC ansatz.
+
+        This is used to ensure that the :attr:`excitation_list` matches with the UCC ansatz that
+        will be used with the VQE algorithm.
+
+        Raises:
+            QiskitNatureError: If not set using a valid
+                :class:`~qiskit_nature.circuit.library.ansatzes.ucc.UCC` instance.
+        """
+        return self._ansatz
+
+    @ansatz.setter
+    def ansatz(self, ansatz: UCC) -> None:
+        if not isinstance(ansatz, UCC):
+            raise QiskitNatureError(f"Expected a UCC ansatz, but got type: {type(ansatz)}.")
+
+        # Operators must be built early to compute the excitation list.
+        _ = ansatz.operators
+
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._excitation_list = ansatz.excitation_list
+        self._ansatz = ansatz
+
+    @property
+    def excitation_list(self) -> list[tuple[tuple[int, ...], tuple[int, ...]]]:
+        """The list of excitations.
+
+        Setting this will overwrite the excitation list from the ansatz.
+        """
+        return self._excitation_list
+
+    @excitation_list.setter
+    def excitation_list(self, excitations: list[tuple[tuple[int, ...], tuple[int, ...]]]):
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._excitation_list = excitations
+
+    def to_numpy_array(self) -> np.ndarray:
+        """The initial point as an array."""
+        if self._corrections is None:
+            self.compute()
+        return np.asarray([value[0] for value in self._corrections.values()])
+
+    def get_energy_corrections(self) -> np.ndarray:
+        """The energy corrections for each excitation."""
+        if self._corrections is None:
+            self.compute()
+        return np.asarray([value[1] for value in self._corrections.values()])
+
+    def get_energy_correction(self) -> float:
+        """The overall energy correction."""
+        return self.get_energy_corrections().sum()
+
+    def get_energy(self) -> float:
+        """The absolute energy.
+
+        If the reference energy was not obtained from
+        :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`
+        this will be equal to :meth:`get_energy_correction`.
+        """
+        return self._reference_energy + self.get_energy_correction()
+
+    def compute(
+        self,
+        ansatz: UCC | None = None,
+        grouped_property: GroupedSecondQuantizedProperty | None = None,
+    ) -> None:
+        """Compute the coefficients and energy corrections.
+
+        See :class:`HFInitialPoint` for more information.
```

Comment:
This is basically a self-reference. I think you could simply say "See further up for more information."